main-service/src/main/java/ru/practicum/mainservice/category/model/Category.java

```diff
@@ -0,0 +1,39 @@
+package ru.practicum.mainservice.category.model;
+
+import lombok.*;
+
+import javax.persistence.*;
+
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "categories")
+public class Category {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Как только удаляю переоределенные equals и hashcode и добавляю @EqualsAndHashCode или @Data,  Idea сразу выдает предупреждение: "Using @EqualsAndHashCode for JPA entities is not recommended. It can cause severe performance and memory consumption issues". 
Кроме того, в гайдах рекомендуется переопределять: https://www.baeldung.com/jpa-entity-equality
В курсе была тема про equals и hashcode да и наставник рассказывал и рекомендовал переопределять.
Предлагаю не менять.

Reply:
Идея предупреждает, так как ломбок бывает криво переопределяет эти методы и JPA ломается) Мы не рекомендуем переопределять методы equals и hashcode совсем, так как в курсе об этом не упоминалось и это далеко не всегда надо делать